lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -85,82 +91,369 @@ describe('NxTable', function() {
           </NxTableRow>
         </NxTableHead>
-        <NxTableBody error="Error!"/>
-      </NxTable>,
-      { attachTo: mountContainer }
+        <NxTableBody error="Error!" />
+      </NxTable>
     );
 
-    await act(async () => {
-      component.setProps({
-        children: (
-          <>
-            <NxTableHead>
-              <NxTableRow>
-                <NxTableCell>Foo</NxTableCell>
-              </NxTableRow>
-            </NxTableHead>
-            <NxTableBody error="Error!"/>
-          </>
-        )
+    expect(screen.getByRole('cell')).toHaveAttribute('colSpan', '2');
+
+    rerender(
+      <NxTable>
+        <NxTableHead>
+          <NxTableRow>
+            <NxTableCell>Foo</NxTableCell>
+          </NxTableRow>
+        </NxTableHead>
+        <NxTableBody error="Error!" />
+      </NxTable>
+    );
+
+    await waitFor(() => expect(screen.getByRole('cell')).toHaveAttribute('colSpan', '1'));
+
+    expect(screen.getByRole('cell')).toHaveClass('nx-cell--meta-info');
+    expect(screen.getByRole('cell')).toHaveTextContent('Error!');
+  });
+
+  describe('NxTableBody', function() {
+    it('does not show the emptyMessage when there are children', async function() {
+      render(
+        <NxTable>
+          <NxTableBody emptyMessage="Empty message">
+            <NxTableRow>
+              <NxTableCell>Foo</NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByRole('rowgroup')).not.toHaveTextContent('Empty message');
+    });
+
+    it('does not show the emptyMessage when isLoading', function() {
+      render(<NxTable><NxTableBody emptyMessage="Empty message" isLoading /></NxTable>);
+      expect(screen.getByRole('rowgroup')).not.toHaveTextContent('Empty message');
+    });
+
+    it('does not show the emptyMessage when in error', function() {
+      render(<NxTable><NxTableBody emptyMessage="Empty message" error="Errr" retryHandler={() => { }} /></NxTable>);
+
+      expect(screen.getByRole('rowgroup')).not.toHaveTextContent('Empty message');
+    });
+
+    it('removes the emptyMessage when children are added', async function() {
+      const { rerender } = render(<NxTable><NxTableBody emptyMessage="Empty message"></NxTableBody></NxTable>);
+
+      expect(screen.getByRole('rowgroup')).toHaveTextContent('Empty message');
+
+      rerender(
+        <NxTable>
+          <NxTableBody emptyMessage="Empty message">
+            <NxTableRow>
+              <NxTableCell />
+            </NxTableRow>
+            <NxTableRow>
+              <NxTableCell />
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      await waitFor(() => expect(screen.getByRole('rowgroup')).not.toHaveTextContent('Empty message'));
+    });
+  });
+
+  describe('NxTableRow', function() {
+    it('sets the rendered text content into RowContext provider label with cells separated by semi-colons', function() {
```

Comment:
The context is an implementation detail.  Instead of interacting with it directly, test that the two features that it is intended to implement do work.  Those features being
1. That filter header cells end up as `<td>` elements rather than `<th>` elements. (or in terms of roles, `cell`s rather than `columnheader`s
2. That the accessible name on the row-click button is correct.

Reply:
> The context is an implementation detail. Instead of interacting with it directly, test that the two features that it is intended to implement do work. Those features being
> 
> 1. That filter header cells end up as `<td>` elements rather than `<th>` elements. (or in terms of roles, `cell`s rather than `columnheader`s
> 2. That the accessible name on the row-click button is correct.

Thanks, Fixed that as part of: c54ac41c226c00680630c11145c503fe66946f46